networkx/algorithms/summarization.py

```diff
@@ -210,3 +213,536 @@ def dedensify(G, threshold, prefix=None, copy=True):
+
+    A group is AR-compatible when all of the nodes in the group have the
+    same node attributes, and all nodes have relationships with the same
+    other groups.
+
+    Parameters
+    ----------
+    group_ids: iterable
+        Group ids of the current graph summary
+    group_sets: iterable
+        Group membership of nodes in the current graph summary
+    participation_counts: dict
+        ID of group
+    group_id: int
+        ID of group
+
+    Returns
+    -------
+    boolean: Indicates if the group can/should be split into multiple groups
+    """
+    group_size = len(group_sets[group_id])
+    for other_group_id in group_ids:
+        group_relationships = participation_counts[group_id][other_group_id]
+        if any(
+            relationship_count != 0 and relationship_count != group_size
+            for relationship_count in group_relationships
+        ):
+            return True
+    return False
+
+
+def snap_split(
+    G,
+    group_ids,
+    group_sets,
+    relationship_attributes,
+    bitmap,
+    participation_counts,
+    group_lookup,
+    group_id=None,
+):
+    """
+    Splits the group with the given group_id based on the relationships
+    of the nodes so that each new grouping will all have the same
+    relationships with other nodes.
+
+    Parameters
+    ----------
+    G: graph
+        Networkx Graph to be summarized
+    group_ids: iterable
+        Group ids of the current graph summary
+    group_sets: iterable
+        Group membership of nodes
+
+    Returns:
+        None
+    """
+    if not group_id:
+        eligible_group_ids = set(
+            group_id
+            for group_id in group_ids
+            if snap_is_eligible(group_ids, group_sets, participation_counts, group_id)
+        )
+        group_id = eligible_group_ids.pop()
+
+    logger.debug("Splitting group %r: %r", group_id, group_sets[group_id])
+    new_group_mappings = dict()
+    for node in group_sets[group_id]:
+        signature = []
+        for grouping in bitmap[node]:
+            signature += grouping
+        signature = tuple(signature)
+        new_group_mappings.setdefault(signature, set())
+        new_group_mappings[signature].add(node)
+    new_groups = list(new_group_mappings.values())
+    logger.debug(
+        "Split group %i into %i groups: %r", group_id, len(new_groups), new_groups
+    )
+    return snap_update_groups(
+        group_ids,
+        group_sets,
+        relationship_attributes,
+        group_id,
+        new_groups[1:],
+        group_lookup,
+    )
+
+
+def snap_aggregation(
+    G, node_attributes, relationship_attributes=None, prefix="Supernode-"
+):
+    """
+    The Summarization by Grouping Nodes on Attributes and Pairwise
+    Relationships (SNAP) algorithm for summarizing a given graph by grouping
+    nodes by node attributes and their edge attributes into supernodes in a
+    summary graph.  Not to be confused with the Stanford Network Analysis
+    Project (SNAP).
```

Comment:
```suggestion
    """Creates a summary graph based on attributes and connectivity.
    
    This function uses the Summarization by Grouping Nodes on Attributes
    and Pairwise Relationships (SNAP) algorithm for summarizing a given
    graph by grouping nodes by node attributes and their edge attributes
    into supernodes in a summary graph.  This name SNAP should not be 
    confused with the Stanford Network Analysis Project (SNAP).
```

Reply:
Good idea to add the steps.  I just added them.